File: src/cc/hashtable/CountingElements.cc
#include <vector>
#include <set>

using namespace std;

/**
 * LeetCode ???. Counting Elements
 */
class Solution {
public:
    int countElements(vector<int>& arr) {
        int cnt = 0;
        set<int> aSet = set<int>();
        for (int i = 0; i < arr.size(); i++) {
            aSet.insert(arr[i]);
        }
        for (int i = 0; i < arr.size(); i++) {
            if (aSet.find(arr[i] + 1) != aSet.end()) {
                cnt++;
            }
        }
        return cnt;
    }
};
int main(int argc, char const *argv[]) {
    // ...
    return 0;
}
/* EOF */

File: src/io/ziheng/hashtable/leetcode/CountingElements.java
package io.ziheng.hashtable.leetcode;

import java.util.HashSet;
import java.util.Set;

/**
 * LeetCode ???. Counting Elements
 */
public class CountingElements {
    public int countElements(int[] arr) {
        if (arr == null || arr.length == 0) {
            return 0;
        }
        int cnt = 0;
        Set<Integer> aSet = new HashSet<>();
        for (int n : arr) {
            aSet.add(n);
        }
        for (int n : arr) {
            if (aSet.contains(n + 1)) {
                cnt++;
            }
        }
        return cnt;
    }
}
/* EOF */